azurerm/internal/services/storage/data_source_storage_container.go

```diff
@@ -0,0 +1,106 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmStorageContainer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmStorageContainerRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"storage_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"container_access_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"metadata": MetaDataComputedSchema(),
+
+			// TODO: support for ACL's, Legal Holds and Immutability Policies
+			"has_immutability_policy": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"has_legal_hold": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmStorageContainerRead(d *schema.ResourceData, meta interface{}) error {
+	storageClient := meta.(*clients.Client).Storage
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	containerName := d.Get("name").(string)
+	accountName := d.Get("storage_account_name").(string)
+
+	account, err := storageClient.FindAccount(ctx, accountName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Account %q for Container %q: %s", accountName, containerName, err)
+	}
+	if account == nil {
+		log.Printf("[DEBUG] Unable to locate Account %q for Storage Container %q - assuming removed & removing from state", accountName, containerName)
+		d.SetId("")
+		return fmt.Errorf("Unable to locate Account %q for Storage Container %q", accountName, containerName)
+	}
+
+	client, err := storageClient.ContainersClient(ctx, *account)
+	if err != nil {
+		return fmt.Errorf("Error building Containers Client for Storage Account %q (Resource Group %q): %s", accountName, account.ResourceGroup, err)
+	}
+
+	d.SetId(client.GetResourceID(accountName, containerName))
+
+	props, err := client.GetProperties(ctx, accountName, containerName)
+	if err != nil {
+		if utils.ResponseWasNotFound(props.Response) {
+			log.Printf("[DEBUG] Container %q was not found in Account %q / Resource Group %q - assuming removed & removing from state", containerName, accountName, account.ResourceGroup)
+			d.SetId("")
+			return fmt.Errorf("Container %q was not found in Account %q / Resource Group %q", containerName, accountName, account.ResourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving Container %q (Account %q / Resource Group %q): %s", containerName, accountName, account.ResourceGroup, err)
+	}
+
+	d.Set("name", containerName)
+
+	d.Set("storage_account_name", accountName)
+
+	d.Set("resource_group_name", account.ResourceGroup)
```

Comment:
since this field is no longer in the schema we can remove this:

```suggestion
```

since trying to set this field when it's not present in the schema causes a panic during the test run:

```
------- Stdout: -------
=== RUN   TestAccDataSourceArmStorageContainer_basic
=== PAUSE TestAccDataSourceArmStorageContainer_basic
=== CONT  TestAccDataSourceArmStorageContainer_basic
panic: Invalid address to set: []string{"resource_group_name"}
goroutine 412 [running]:
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).Set(0xc00052ad90, 0x412dce7, 0x13, 0x3a51980, 0xc000bec0c0, 0x0, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.1/helper/schema/resource_data.go:193 +0x354
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage.dataSourceArmStorageContainerRead(0xc00052ad90, 0x3b46520, 0xc00063ce00, 0x0, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/data_source_storage_container.go:94 +0xbbf
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc0007ede80, 0xc0006d2140, 0x3b46520, 0xc00063ce00, 0xc0005390d0, 0x1, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.1/helper/schema/resource.go:398 +0x88
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc000715900, 0xc000c279d8, 0xc0006d2140, 0xc0006d2140, 0x0, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.1/helper/schema/provider.go:451 +0x8f
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc000d18760, 0x48c3440, 0xc0012dd3e0, 0xc0002482c0, 0xc000d18760, 0xc0012dd3e0, 0xc00008ea80)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.1/internal/helper/plugin/grpc_provider.go:1036 +0x44e
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler(0x3ef9940, 0xc000d18760, 0x48c3440, 0xc0012dd3e0, 0xc000925f20, 0x0, 0x48c3440, 0xc0012dd3e0, 0xc000cecb40, 0xbc)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.1/internal/tfplugin5/tfplugin5.pb.go:3225 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0013e54a0, 0x49167a0, 0xc001142d80, 0xc000c4ce00, 0xc0010fd500, 0x7324990, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:995 +0x460
google.golang.org/grpc.(*Server).handleStream(0xc0013e54a0, 0x49167a0, 0xc001142d80, 0xc000c4ce00, 0x0)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1275 +0xd97
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0010cacd0, 0xc0013e54a0, 0x49167a0, 0xc001142d80, 0xc000c4ce00)
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/dbea4b84960ec72a/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xa1
FAIL	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/tests	44.693s
```

Reply:
I've re-run the test and here is the test result on my side 

```
GOROOT=/usr/local/go #gosetup
GOPATH=/Users/virgil/go #gosetup
/usr/local/go/bin/go test -c -o /private/var/folders/bm/tdy2xbzx74s7fj72gsqwt7sr0000gn/T/___TestAccDataSourceArmStorageContainer_basic_in_github_com_terraform_providers_terraform_provider_azurerm_azurerm_internal_services_storage_tests github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/tests #gosetup
/usr/local/go/bin/go tool test2json -t /private/var/folders/bm/tdy2xbzx74s7fj72gsqwt7sr0000gn/T/___TestAccDataSourceArmStorageContainer_basic_in_github_com_terraform_providers_terraform_provider_azurerm_azurerm_internal_services_storage_tests -test.v -test.run ^TestAccDataSourceArmStorageContainer_basic$ #gosetup
=== RUN   TestAccDataSourceArmStorageContainer_basic
=== PAUSE TestAccDataSourceArmStorageContainer_basic
=== CONT  TestAccDataSourceArmStorageContainer_basic
--- PASS: TestAccDataSourceArmStorageContainer_basic (125.41s)
PASS

Process finished with exit code 0
```